### ISSUE TITLE
I expect "help" to be a subcommand

### ISSUE THREAD

**Issue by cwellsx** [dotnet/command-line-api#1417]
When a root command has no subcommands then "help" is supported - I guess it's an implicitly-supported argument or subcommand:

    foo help

When there are subcommands (e.g. `bar`) then the user requests help as an option (one of `-?` or `-h` or `--help`):

    foo bar --help

But as a user I expect that "help" should be a subcommand e.g. like this:

    foo help bar

Git for example supports this:

    git help fetch

So I suggest you support it too, except in the case where the programmer has explicitly defined "help" as a subcommand.

Or I'd like a code snippet/sample to let me define that help subcommand myself -- i.e. to get the help text associated with other defined subcommands.

**Comment by elgonzo** [comment]
> When a root command has no subcommands then "help" is supported - I guess it's an implicitly-supported argument or subcommand:

No, it's not, at least not with recent System.CommandLine versions. If "help" is not a subcommand you have defined, then "help" will be an _unknown_ (sub)command, and using it will provide an **error message** plus help text. You would also get a similar error message and help text if you were to use any other undefined (sub)command like for example `foo ThisCmdDoesNotExist`.

You could define individual _help_ sub-commands for each of your commands, with the command handler for each of those _help_ sub-commands producing (and outputting) the help-text for the respective command with an `HelpBuilder`-based approach similar to what is being demonstrated here: https://github.com/dotnet/command-line-api/issues/1392#issuecomment-909143059

While you could add each _help_ subcommand manually, it probably is better to add those _help_ subcommands programmatically after creating your otherwise "ordinary" command instances. Keep in mind that a `Command` instance is also an `IEnumerable<Symbol>` -- an enumerable that enumerates all options, arguments, and sub-commands of that Command instance. Thus, given a command, you can iterate over it like `foreach (var subCmd in myCmd.OfType<Command>()) ...`, and then create a _help_ subcommand with a command handler that creates the help text for _subCmd_:
```C#
private void AddHelpSubCommandsRecursively(Command cmd)
{
    foreach (var subCmd in cmd.OfType<Command>())
    {
        AddHelpSubCommandsRecursively(subCmd);
    }

    var localSubCmdVar = subCmd;
    var helpCmd = new Command("help")
    {
        Handler = CommandHandler.Create(() => myHelpBuilder.Write(localSubCmdVar))
    };
    subCmd.AddCommand(helpCmd);
}
```
```C#
AddHelpSubCommands(myRootCmd);
```
_Warning! The code above is unverified and untested, just from the top of my head. But i hope that even in case it containing errors it helps illustrating an approach of how you might achieve your goal..._
